sdks/cpp/lite/include/Device.h

```diff
@@ -146,6 +147,140 @@ class Device {
      */
     void toProto(::catena::LanguageList& list) const;
 
+    /**
+     * @brief DeviceSerializer is a coroutine that serializes the device into a stream of DeviceComponents
+     * This struct manages the state and lifetime of the coroutine. It also provides the interface for resuming the coroutine.
+     */
+    struct DeviceSerializer {
+        struct promise_type; // forward declaration
+
+      private:
+        /**
+         * The coroutine handle is a pointer to the coroutine state
+         * 
+         * The coroutine handle provides the following operations:
+         * resume() - resumes the coroutine from where it was suspended
+         * destroy() - destroys the coroutine
+         * done() - returns true if the coroutine has completed (i.e. co_return has been called)
+         */
+        using handle_type = std::coroutine_handle<promise_type>;
+        handle_type handle_;
+
+      public:
+        /** 
+         * @brief promise_type defines the execution behaviour of the coroutine
+         * 
+         * Coroutine types are required to contain a promise_type struct that defines some special
+         * member functions that the compiler will call when the coroutine is created, destroyed,
+         * or resumed.
+         */
+        struct promise_type {
+
+            /**
+             * get_return_object() is called when the coroutine is created (i.e. when getComponentSerializer() 
+             * is called). It creates the handle for the coroutine and returns the DeviceSerializer object.
+             */
+            DeviceSerializer get_return_object() { 
```

Comment:
should be declared inline or defined in a .cpp file, same comment applied to other methods defined here. recommend inlining them as they're all v short.